index.js

```
@@ -0,0 +1,128 @@
+
+      // Get elements containing info about each class within each day in the schedule
+      const classElements = scheduleDayElement.querySelectorAll('ul li');
+
+      // If the schedule day doesn't have any classes, don't add classes to accumulator
+      if (!classElements.length) {
+        return acc;
+      }
+
+      // Format classes for ics
+      const classes = [...classElements].map((classElement) => {
+        // Info about each class
+        const className = classElement.querySelector('.class-name').textContent.trim();
+        assert(className.length);
+        const classTimes = classElement.querySelector('.time').textContent.trim().split(' - ').map((classTimeString) => {
+          const classTime = DateTime.fromFormat(classTimeString, 'h:mma')
+          assert(classTime.isValid);
+          return classTime;
+        });
+        const classStartTime = classTimes[0];
+        const classEndTime = classTimes[1];
+        assert(classTimes.length === 2);
+        const classRoom = classElement.querySelector('.room').textContent.trim() || '';
+        const classHref = `${scheduleURL.origin}${classElement.querySelector('a').href}` || '';
+        const classDescription = `Details: ${classHref !== scheduleURL.origin ? classHref : scheduleURL.href}`;
+
+        return {
+          title: className,
+          location: classRoom,
+          description: classDescription,
+          start: [scheduleDate.year, scheduleDate.month, scheduleDate.day, classStartTime.hour, classStartTime.minute],
+          end: [scheduleDate.year, scheduleDate.month, scheduleDate.day, classEndTime.hour, classEndTime.minute],
+        };
+      });
+
+      // Add classes to accumulator
+      return acc.concat(classes);
+    }, []);
+
+    // Create ics-formatted data
+    ics.createEvents(events, (error, value) => {
+      if (error) {
+        console.error(error)
```

Consider aborting at this point instead of just printing it. You may want to just `assert(!error)`? I don't know what kind of errors this might come out with, but you probably want an error exit code and no output, at least, to match your other errors.